babel/src/main/java/org/apache/calcite/sql/babel/SqlAlterTableAddColumns.java

```diff
@@ -20,28 +20,28 @@
 import org.apache.calcite.sql.SqlNodeList;
 import org.apache.calcite.sql.SqlWriter;
 
+import java.util.Objects;
+
 /**
- * A {@code SqlAlterTableAddColumn} represents
- * an ADD column statement within an ALTER TABLE query.
+ * A {@code SqlAlterTableAddColumn} represents an ADD column statement within
+ * an ALTER TABLE query.
  */
-public class SqlAlterTableAddColumn extends SqlAlterTableOption {
+public class SqlAlterTableAddColumns extends SqlAlterTableOption {
 
   public final SqlNodeList columns;
 
-  public SqlAlterTableAddColumn(SqlNodeList columns) {
-    this.columns = columns;
+  public SqlAlterTableAddColumns(SqlNodeList columns) {
+    this.columns = Objects.requireNonNull(columns);
```

Comment:
Should we also assert that this is not an empty list using SqlNodeList.isEmptyList()?

Reply:
Sure, that makes sense. I guess the only situation where an empty list would be passed into the constructor would be if someone wanted to instantiate a SqlAlterTableAddColumns object and add columns to the list incrementally, but I don't think there is a situation where that would be necessary, so it's probably safe to assume that if the list is empty then it is invalid.

Reply:
Actually, this made me think of another question: should we be supporting the use case of modifying "columns" after the SqlAlterTableAddColumns has been created?

Should we instead make a copy of columns in the constructor so that any modifications that happen after instantiation don't mutate this class' state?

Reply:
I have actually wondered about this as well. There are many places throughout the code (for example in SqlCreateTable, SqlExecMacro, SqlSelect) where a list is passed in and stored directly, without making a copy, and could therefore be modified later on, outside the class. We could make a copy here to ensure the internal state of SqlAlterTableAddColumns can't be changed, but this is really a much larger issue that isn't limited to this class. If this needs to be addressed, it might be worth opening a new refactoring feature (something along the lines of "ensuring object state immutability") that tackles this on a larger scale.

Reply:
That sounds reasonable to me. I think it's probably fine to leave for now.